plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/SolverInteraction.kt

```diff
@@ -130,3 +130,16 @@ internal fun SolverState.checkConditionsInconsistencies(
       MetaErrors.InconsistentConditions.on(expression.psiOrParent, unsatCore)
     )
   }
+
+internal fun SolverState.checkInvariant(
+  formula: BooleanFormula?,
```

Comment:
We should also do a pass in another PR to ensure we are not taking nullable values as arguments as it's easier to delegate the responsibility at the call site to only invoke the function if there is an actual formula. It should simplify all functions where we return defaults based on nullable arguments.

Reply:
Done